cmd/config/edit.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package config
+
+import (
+	"os"
+
+	"github.com/k0sproject/k0s/pkg/config"
+	"github.com/spf13/cobra"
+)
+
+func NewEditCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "edit",
+		Short: "Launch the editor configured in your shell to edit k0s configuration",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			c := CmdOpts(config.GetCmdOpts())
+			os.Args = []string{"k0s", "kubectl", "--data-dir", c.K0sVars.DataDir, "-n", "kube-system", "edit", "clusterconfig", "k0s"}
```

Comment:
Shouldn't this take the path to k0s from `os.Executable()`?

Reply:
It doesn't exec the binary, just fakes the os.Args for cmd.Execute (which only uses Args[1:])

But yes maybe it could reuse Args[0]

Reply:
`cmd.Execute()` calls `c.Parent().Root()`, that's how it ends up in the kubectl command.

Reply:
oh, right. somehow my brain interpreted the `cmd.Execute` as `os.Execute`... 🤦